modules/rubiconBidAdapter.js

```diff
@@ -176,166 +286,57 @@ export const spec = {
-      bidRequest.startTime = new Date().getTime();
-
-      const data = {
-        id: bidRequest.transactionId,
-        test: config.getConfig('debug') ? 1 : 0,
-        cur: ['USD'],
-        source: {
-          tid: bidRequest.transactionId
-        },
-        tmax: bidderRequest.timeout,
-        imp: [{
-          exp: config.getConfig('s2sConfig.defaultTtl'),
-          id: bidRequest.adUnitCode,
-          secure: 1,
-          ext: {
-            [bidRequest.bidder]: bidRequest.params
-          },
-          video: deepAccess(bidRequest, 'mediaTypes.video') || {}
-        }],
-        ext: {
-          prebid: {
-            channel: {
-              name: 'pbjs',
-              version: $$PREBID_GLOBAL$$.version
-            },
-            cache: {
-              vastxml: {
-                returnCreative: rubiConf.returnVast === true
-              }
-            },
-            targeting: {
-              includewinners: true,
-              // includebidderkeys always false for openrtb
-              includebidderkeys: false,
-              pricegranularity: getPriceGranularity(config)
-            },
-            bidders: {
-              rubicon: {
-                integration: rubiConf.int_type || DEFAULT_PBS_INTEGRATION
-              }
-            }
-          }
-        }
-      }
-
-      // Add alias if it is there
-      if (bidRequest.bidder !== 'rubicon') {
-        data.ext.prebid.aliases = {
-          [bidRequest.bidder]: 'rubicon'
-        }
-      }
-
-      let modules = (getGlobal()).installedModules;
-      if (modules && (!modules.length || modules.indexOf('rubiconAnalyticsAdapter') !== -1)) {
-        deepSetValue(data, 'ext.prebid.analytics', {'rubicon': {'client-analytics': true}});
-      }
-
-      let bidFloor;
-      if (typeof bidRequest.getFloor === 'function' && !rubiConf.disableFloors) {
-        let floorInfo;
-        try {
-          floorInfo = bidRequest.getFloor({
-            currency: 'USD',
-            mediaType: 'video',
-            size: parseSizes(bidRequest, 'video')
-          });
-        } catch (e) {
-          logError('Rubicon: getFloor threw an error: ', e);
-        }
-        bidFloor = typeof floorInfo === 'object' && floorInfo.currency === 'USD' && !isNaN(parseInt(floorInfo.floor)) ? parseFloat(floorInfo.floor) : undefined;
-      } else {
-        bidFloor = parseFloat(deepAccess(bidRequest, 'params.floor'));
-      }
-      if (!isNaN(bidFloor)) {
-        data.imp[0].bidfloor = bidFloor;
-      }
-
-      // If the price floors module is active, then we need to signal to PBS! If floorData obj is present is best way to check
-      if (typeof bidRequest.floorData === 'object') {
-        data.ext.prebid.floors = { enabled: false };
-      }
-
-      // if value is set, will overwrite with same value
-      data.imp[0].ext[bidRequest.bidder].video.size_id = determineRubiconVideoSizeId(bidRequest)
```

Comment:
Looks like this is not being used and needs to be.

If user does not pass in `params.size_id` than it should determine what to pass in bidder params using this function.

So just add somewhere in the `imp` processor to do this.

Reply:
Hmm I'm not sure I understand your point, because video is a mistery to me; anyway, I received an input from @bretg to remove this size_id param : you can find the note in the issue https://magnite.atlassian.net/browse/HB-15147 

> imp.ext.rubicon.video.size_id - the server-side adapter will set sizeid

Reply:
👍